stdlib/format.mli

```diff
@@ -107,6 +103,446 @@
+
+{b When defining the box}: when you open a box, you can fix the indentation
+added to each new line opened within that box.
+
+For instance: [open_hovbox 1] opens a "hov" box with new lines indented 1 more
+than the initial indentation of the box. With output "---\[--b--b--b--", we
+get:
+
+{[
+---\[--b--b
+     --b--
+]}
+
+with open_hovbox 2, we get
+
+{[
+---\[--b--b
+      --b--
+]}
+
+Note: the \[ sign in the display is not visible on the screen, it is just there
+to materialise the aperture of the pretty-printing box. Last "screen" stands
+for:
+
+{[
+-----b--b
+     --b--
+]}
+
+{b When defining the break that makes the new line}. As said above, you output
+a break hint using [print_break sp indent]. The [indent] integer is used to fix
+the additional indentation of the new line. Namely, it is added to the default
+indentation offset of the box where the break occurs.
+
+For instance, if \[ stands for the opening of a "hov" box with 1 as extra
+indentation (as obtained by [open_hovbox 1]), and b is [print_break 1 2], then
+  from output "---\[--b--b--b--", we get:
+
+{[
+   ---\[-- --
+         --
+         --
+]}
+
+{2 Refinement on "hov" boxes}
+
+
+The "hov" box type is refined into two categories.
+
+- {b the vertical or horizontal {i packing} box} (as obtained by the
+{!open_hovbox} procedure): break hints are used to cut the line when there is no
+more room on the line; no new line occurs if there is enough room on the line.
+- {b vertical or horizontal {i structural} box} (as obtained by the {!open_box}
+procedure): similar to the "hov" packing box, the break hints are used to cut
+the line when there is no more room on the line; in addition, break hints that
+can show the box structure lead to new lines even if there is enough room on
+the current line.
+
+
+The difference between a packing and a structural "hov" box is shown by a
+routine that closes boxes and parentheses at the end of printing: with packing
+boxes, the closure of boxes and parentheses do not lead to new lines if there
+is enough room on the line, whereas with structural boxes each break hint will
+lead to a new line. For instance, when printing
+"\[(---\[(----\[(---b)\]b)\]b)\]", where "b" is a break hint without extra
+indentation ([print_cut ()]). If "\[" means opening of a packing "hov" box
+({!open_hovbox}), "\[(---\[(----\[(---b)\]b)\]b)\]" is printed as follows:
+
+{[
+(---
+ (----
+  (---)))
+]}
+
+If we replace the packing boxes by structural boxes ({!open_box}), each break
+hint that precedes a closing parenthesis can show the boxes structure, if it
+leads to a new line; hence "\[(---\[(----\[(---b)\]b)\]b)\]" is printed like
+this:
+
+{[
+(---
+ (----
+  (---
+  )
+ )
+)
+]}
+
+{2 Practical advice}
+
+When writing a pretty-printing routine, follow these simple rules:
+
++ Boxes must be opened and closed consistently ([open_*] and {!close_box} must
+be nested like parentheses).
++ Never hesitate to open a box.
```

Comment:
This advice does not work in my experience. It may lead people to write the following `list` function:
```ocaml
let rec list elt ppf = function
| [] -> ()
| a :: q -> Format.fprintf ppf "%a,@ @[%a@]" elt a (list elt) q
```